torchvision/prototype/datasets/_builtin/dtd.py

```diff
@@ -0,0 +1,130 @@
+import io
+import pathlib
+from typing import Any, Callable, Dict, List, Optional, Tuple
+
+import torch
+from torchdata.datapipes.iter import (
+    IterDataPipe,
+    Mapper,
+    Shuffler,
+    Filter,
+    IterKeyZipper,
+    Demultiplexer,
+    LineReader,
+    CSVParser,
+)
+from torchvision.prototype.datasets.utils import (
+    Dataset,
+    DatasetConfig,
+    DatasetInfo,
+    HttpResource,
+    OnlineResource,
+    DatasetType,
+)
+from torchvision.prototype.datasets.utils._internal import (
+    INFINITE_BUFFER_SIZE,
+    hint_sharding,
+    path_comparator,
+    getitem,
+)
+from torchvision.prototype.features import Label
+
+
+class DTD(Dataset):
+    def _make_info(self) -> DatasetInfo:
+        return DatasetInfo(
+            "dtd",
+            type=DatasetType.IMAGE,
+            homepage="https://www.robots.ox.ac.uk/~vgg/data/dtd/",
+            valid_options=dict(
+                split=("train", "test", "val"),
+                fold=tuple(str(fold) for fold in range(1, 11)),
+            ),
+        )
+
+    def resources(self, config: DatasetConfig) -> List[OnlineResource]:
+        archive = HttpResource(
+            "https://www.robots.ox.ac.uk/~vgg/data/dtd/download/dtd-r1.0.1.tar.gz",
+            sha256="e42855a52a4950a3b59612834602aa253914755c95b0cff9ead6d07395f8e205",
+            decompress=True,
+        )
+        return [archive]
+
+    def _classify_archive(self, data: Tuple[str, Any]) -> Optional[int]:
+        path = pathlib.Path(data[0])
+        if path.parent.name == "labels":
+            if path.name == "labels_joint_anno.txt":
+                return 1
+
+            return 0
+        elif path.parents[1].name == "images":
+            return 2
```

Comment:
instead of hardcoded 0 1 2, would a small private enum be overkill here? or maybe just hardcoded named constants? No strong opinion but this might help readability

Reply:
Yes, that could work. If that is ok with you, I'll put it in my backlog, because we should do that for all datasets that use a `Demultiplexer`.